src/main/java/duke/models/tasks/TaskManager.java

```diff
@@ -60,18 +60,30 @@ public TaskManager() {
         return tasksWithThisDescription;
     }
 
-
     /**
      * Adds a Task to the list.
      *
-     * @param task The Task to be added to the list.
+     * @param taskToBeAdded contains the Task to be added to the list.
+     * @throws DukeException if the task is not added successfully.
      */
-    public void addTask(Task task) {
-        if (task.getId() == 0) {
+    public void addTask(Task taskToBeAdded) throws DukeException {
+        if (taskToBeAdded.getDescription().length() < 4) {
+            throw new DukeException("The task description is too short.");
+        }
+        ArrayList<Task> sameTasks = new ArrayList<>();
+        for (Task task : taskIdMap.values()) {
```

Comment:
Iterating through map values can be damaging if you want efficiency (so the program will probably take longer to run), but it's something we can optimize if we have more time later.

Reply:
Good point. Will look into this problem in the future.